model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -633,18 +629,13 @@ def time_step(
             at_last_substep=at_last_substep,
         )
 
-        start_cell_lb = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.CellDim)
-        )
-        end_cell_nudging_minus1 = self.grid.get_end_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.nudging(dims.CellDim) - 1
-        )
-        start_cell_halo = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.halo(dims.CellDim)
-        )
-        end_cell_end = self.grid.get_end_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.end(dims.CellDim)
-        )
+        start_cell_lb = self.grid.start_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY))
+
+        end_cell_lb_level_4 = self.grid.end_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_4))
```

Comment:
More consistency renaming

```suggestion
        end_cell_lateral_boundary_level_4 = self.grid.end_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_4))
```